File: Service/Admin/RMA.Service.Admin.CampaignManager.Contracts/Entities/Campaign.cs
using RMA.Common.Entities;
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

using System;
using System.Collections.Generic;

namespace RMA.Service.Admin.CampaignManager.Contracts.Entities
{
    public class Campaign : AuditDetails
    {
        public Campaign()
        {
            TargetAudiences = new List<TargetAudience>();
            CampaignEmails = new List<Email>();
            CampaignSmses = new List<Sms>();
        }

        public int TenantId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public CampaignCategoryEnum CampaignCategory { get; set; }
        public CampaignTypeEnum CampaignType { get; set; }
        public CampaignStatusEnum CampaignStatus { get; set; }
        public int? ProductId { get; set; }
        public string Owner { get; set; }
        public string Role { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public bool Paused { get; set; }
        public DateTime? DateViewed { get; set; }
        public List<TargetAudience> TargetAudiences { get; set; }
        public List<Email> CampaignEmails { get; set; }
        public List<Sms> CampaignSmses { get; set; }
        public string Note { get; set; }
        public DateTime WeekEnding { get; set; }
        public DateTime CollectionDate { get; set; }
        public decimal CollectionAmount { get; set; }
        public string PhoneNumber { get; set; }
        public string EmailAddress { get; set; }

        public override string ToString()
        {
            return $"Campaign: {Id} {Name}";
        }
    }
}

File: Web/RMA-Web-Shell/projects/admin/src/app/campaign-manager/shared/entities/campaign.ts
import { TargetAudience } from './target-audience';
import { CampaignEmail } from './campaign-email';
import { BaseClass } from 'projects/shared-models-lib/src/lib/common/base-class';
import { CampaignSms } from './campaign-sms';

export class Campaign extends BaseClass {
    tenantId: number;
    name: string;
    description: string;
    campaignCategory: number;
    campaignType: number;
    campaignStatus: number;
    productId: number;
    owner: string;
    role: string;
    startDate: Date;
    endDate: Date;
    dateViewed: Date;
    paused: boolean;

    targetAudiences: TargetAudience[];
    campaignEmails: CampaignEmail[];
    campaignSmses: CampaignSms[];
    // Optional fields for billing campaigns
    recipient: string;
    note: string;
    weekEnding: Date;
    collectionDate: Date;
    collectionAmount: number;
    phoneNumber: string;
    emailAddress: string;
    phoneTarget: any;
    emailTarget: any;
}
